app/src/main/java/ch/epfl/skysync/screens/UserManagement.kt

```diff
@@ -0,0 +1,260 @@
+fun TopBarTitle(userCount: Int) {
+  Row(verticalAlignment = Alignment.CenterVertically) {
+    Text(
+        text = "Users",
+        style = MaterialTheme.typography.headlineLarge,
+        modifier = Modifier.padding(end = 8.dp))
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier.background(color = lightOrange, shape = CircleShape).padding(8.dp)) {
+          Text(
+              text = userCount.toString(),
+              style = MaterialTheme.typography.titleLarge,
+          )
+        }
+  }
+}
+
+// Composable function for the search bar.
+@Composable
+fun SearchBar(query: String, onQueryChanged: (String) -> Unit) {
+  OutlinedTextField(
+      value = query,
+      onValueChange = onQueryChanged,
+      modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 8.dp),
+      placeholder = { Text("Search users") },
+      singleLine = true,
+      trailingIcon = {
+        if (query.isNotEmpty()) {
+          IconButton(onClick = { onQueryChanged("") }) {
+            Icon(imageVector = Icons.Filled.Clear, contentDescription = "Clear search query")
+          }
+        }
+      })
+}
+
+// Composable function to filter users by role.
+@Composable
+fun RoleFilter(onRoleSelected: (RoleType?) -> Unit, roles: List<RoleType>) {
+  var expanded by remember { mutableStateOf(false) } // State to manage dropdown expansion.
+  var displayText by remember { mutableStateOf("Filter by role") }
+
+  Column(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+    Button(
+        onClick = { expanded = true },
+        colors =
+            ButtonDefaults.buttonColors(containerColor = lightOrange, contentColor = Color.Black)) {
+          Text(text = displayText)
+          Icon(
+              imageVector = Icons.Default.ArrowDropDown,
+              contentDescription = "Expand or collapse menu")
+        }
+
+    // Dropdown menu for selecting roles.
+    DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
+      DropdownMenuItem(
+          onClick = {
+            displayText = "Filter by role"
+            onRoleSelected(null)
+            expanded = false
+          },
+          text = { Text("All Roles") })
+      roles.forEach { role ->
+        DropdownMenuItem(
+            onClick = {
+              displayText = role.name
+              onRoleSelected(role)
+              expanded = false
+            },
+            text = { Text(role.name) })
+      }
+    }
+  }
+}
+
+// Main screen composable integrating all components.
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun UserManagementScreen(navController: NavHostController) {
+  var searchQuery by remember { mutableStateOf("") }
+  var selectedRole by remember { mutableStateOf<RoleType?>(null) }
+  val roles = RoleType.entries
+
+  // Filter users based on search query and selected role.
+  val filteredUsers =
+      mockUsers.filter {

```

Comment:
Avoid utilizing the hardcoded list directly. Instead, you should pass it as an argument to the UserManagementScreen.

Reply:
Yep was for initial preview and is fixed now